### PR TITLE
Manage mentors flow

### PR DESCRIPTION
**Pull request by simonwhatley**

### CODE REVIEW

**Comment by simonwhatley** [comment]
[Add a mentor - design](https://trello.com/c/8cJacZWt/117-add-a-mentor-design)

File: app/controllers/mentors.js [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+  const mentor = mentorModel.findOne({
+    organisationId: req.params.organisationId,
+    trn: req.session.data.mentor.trn
+  })
+
+  if (mentor) {
+    const error = {}
+    error.fieldName = 'mentor'
+    error.href = '#mentor'
+    error.text = 'Mentor has already been added'
```
Comment (LeaWelch) [inline app/controllers/mentors.js:97]: 'Mentor has already been added' isn't as natural as 'The mentor has already been added' (yes it's adding to character count, but as per GDS guidance, errors should sound natural)

File: app/views/mentors/find.njk [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+            classes: "govuk-label--l"
+          },
+          value: data.mentor.trn,
+          classes: "govuk-input--width-20",
+          errorMessage: errors | getErrorMessage("mentor")
+        }) }}
+
+        {{ govukDetails({
+          summaryText: "I don’t know the teacher reference number",
+          html: 'You can use the <a href="https://find-a-lost-trn.education.gov.uk/">Find a lost teacher reference number (TRN)</a> service to find the TRN for the mentor.'
```
Comment (LeaWelch) [inline app/views/mentors/find.njk:43]: html: 'You can use the <a href="https://find-a-lost-trn.education.gov.uk/"> find a lost teacher reference number (TRN)</a> service to find a TRN, or to check the mentor has one. '

File: app/views/mentors/list.njk [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+      {{ govukButton({
+        text: "Add mentor",
+        href: actions.new
+      }) }}
+
+      {% if mentors.length %}
+        {% include "_includes/mentors/list.njk" %}
+      {% else %}
+        <p class="govuk-body">There are no mentors for {{ organisation.name if organisation.name else "your organisation" }}.</p>
```
Comment (LeaWelch) [inline app/views/mentors/list.njk:28]: <p class="govuk-body">There are no mentors to show for {{ organisation.name if organisation.name else "your organisation" }}.</p>

**Comment by LeaWelch** [comment]
@simonwhatley I just added a few comments for consideration. Thanks!

**Comment by simonwhatley** [comment]
Related PR https://github.com/DFE-Digital/itt-mentoring-beta-prototype/pull/32

**Pull request merged**